modules/swagger-jaxrs/src/test/java/io/swagger/SimpleReaderTest.java

```diff
@@ -682,7 +682,7 @@ public void scanClassWithReadOnlyAndEmptyValueParams() {
         Parameter readOnlyParam = swagger.getPath("/fun/readOnly").getGet().getParameters().get(0);
         assertTrue(readOnlyParam.isReadOnly());
 
-        Parameter allowEmptyParam = swagger.getPath("/fun/allowEmpty").getGet().getParameters().get(0);
+        QueryParameter allowEmptyParam = (QueryParameter) swagger.getPath("/fun/allowEmpty").getGet().getParameters().get(0);
```

Comment:
you would use here `SerializableParameter` instead of `QueryParameter`

Reply:
This attribute must be valid only to QueryParameter and FormParameter as the documentation says: ```This is valid only for either query or formData parameters and allows you to send a parameter with a name only or an empty value```.

Reply:
We are moving the property from the more generic interface/abstract class `Parameter/AbstractParameter` which is also implemented by `BodyParameter`, to the less generic `SerializableParameter/AbstractSerializableParameter` which is not implemented by BodyParameter; this is implemented also by other parameters (not query or formData) but it's a good compromise of effort/result for this scope.

Reply:
Now I've understood the change. Thanks! Changed to SerializableParameter.